addons/website_slides/models/slide_channel.py

```diff
@@ -765,6 +765,8 @@ def _remove_membership(self, partner_ids):
 
     def _send_share_email(self, emails):
         """ Share channel through emails."""
+        if not self.share_channel_template_id.id:
+            raise UserError(_("You cannot send emails because of the missing template"))
```

Comment:
This is not sufficient because the method can be called on multiple courses at once. 

1.  This call will fail if courses in `self` have a different template set because `.id` expects a single record. 
2. `self.share_channel_template_id.ids` would technically work, but if one record doesn't have a template attached and others do, the recordset will not be empty and you will not have caught the problem = same internal error)
3. If you "only" check that all courses in `self` have a template set, raising an error with this message will not be explicit for the user. E.g., if sharing for X courses how does the user know how many and which one(s) should they check?

You'll find examples in the code, two in `event` where we handle this differently:
One simple here, where we are explicit at the first error
https://github.com/odoo/odoo/blob/16.0/addons/event/models/event_ticket.py#L146-149

A more detailed here, where we are more helpful :) (the check is simpler in this case).
https://github.com/odoo/odoo/blob/16.0/addons/event/models/event_ticket.py#L153-L161

Could you aim for this kind of message here?
Something like: `Impossible to send emails. Select a "Channel Share Template" for the following courses first: `